src/main/java/com/connexta/search/index/controllers/MultipartFileValidator.java

```diff
@@ -16,47 +16,45 @@
  * Encapsulate common validations of an incoming MultipartFile. Throw a ValidationException if there
  * is a violation.
  */
-
 // TODO Use a proper assertion library like Truth or Hamcrest instead of if statements
 public class MultipartFileValidator {
-  private final MultipartFile file;
+
   private static final long GIGABYTE = 1 << 30;
   private static final long MAX_FILE_BYTES = 10 * GIGABYTE;
 
-  public MultipartFileValidator(MultipartFile file) {
-    this.file = file;
-  }
+  private MultipartFileValidator() {}
 
-  public void validate() {
-    validateSize();
-    validateContentType();
-    validateFilename();
-    validateInputStream();
+  public static MultipartFile validate(final MultipartFile file) {
```

Comment:
You might want to make this an instance method so the validator can be injected as a bean.

Reply:
What would be the value of injecting it in the bean? Would we ever mock it? Is it more efficient?